Show rendered template when invalid yaml

As `--dry-run -o yaml` will only output errors if the file is valid yaml it would still be usefull to get the template result to determin where in a file an error is.
```
bash-4.3# kd -f kube/config-bx.yml -- --dry-run -o yaml
[ERROR] 2018/04/17 11:03:32 main.go:123: yaml: line 28: could not find expected ':'
```
Would like to add a flag to show rendered templates e.g. `--debug-templates`.

--- a/README.md
+++ b/README.md
@@ -77,6 +77,8 @@ You can fail an ongoing deployment if there's been a new deployment by adding `-
 
 ## Templating
 
+You can add the flag --debug-templates to render templates at run time.
+
 ### split
 
 `split` function is go's `strings.Split()`, it returns a `[]string`. A range function

--- a/main.go
+++ b/main.go
@@ -50,6 +50,11 @@ func main() {
 			EnvVar: "DEBUG,PLUGIN_DEBUG",
 		},
 		cli.BoolFlag{
+			Name:   "debug-templates",
+			Usage:  "debug template output",
+			EnvVar: "DEBUG_TEMPLATES,PLUGIN_DEBUG_TEMPLATES",
+		},
+		cli.BoolFlag{
 			Name:   "insecure-skip-tls-verify",
 			Usage:  "if true, the server's certificate will not be checked for validity",
 			EnvVar: "INSECURE_SKIP_TLS_VERIFY,PLUGIN_INSECURE_SKIP_TLS_VERIFY",
@@ -186,6 +191,9 @@ func run(c *cli.Context) error {
 	}
 
 	for _, r := range resources {
+		if c.Bool("debug-templates") {
+			logInfo.Printf("Template:\n" + string(r.Template[:]))
+		}
 		if err := yaml.Unmarshal(r.Template, &r); err != nil {
 			return err
 		}

